src/commons/services/utils.js

```diff
@@ -222,6 +222,62 @@
             return poolList;
         };
 
+        vm.getRBDsDetails = function(clusterId) {
+            var rbdList = [], index1, index2;
+
+            if($rootScope.clusterData !== null) {
+                clusterData = $rootScope.clusterData.clusters;
+                len = clusterData.length;
+
+                for ( i = 0; i < len; i++ ) {
```

Comment:
i don't think there is any other option for doing this ..
because we are not using any third party lib like underscore.js and lodash.js.
all the thing we are doing manually.

please have look the response format :- 

{
	"clusters": [{
		"fsid": "64d779e6-491e-4998-a805-1252bbd5a899",
		"integration_id": "64d779e6-491e-4998-a805-1252bbd5a899",
		"name": "ceph",
		"sds_name": "ceph",
		"sds_version": "10.2.5",
		"cluster_id": "64d779e6-491e-4998-a805-1252bbd5a899",
		"pools": {
			"14": {
				"pool_name": "dummy",
				"type": "replicated",
				"used": "0",
				"erasure_code_profile": "",
				"min_size": "2",
				"percent_used": "0",
				"pg_num": "128",
				"pool_id": "14"
			},
			"4": {
				"min_size": "2",
				"percent_used": "0",
				"type": "replicated",
				"used": "58",
				"rbds": {
					"mmrbd1": {
						"features": "layering, exclusive-lock, object-map, fast-diff, deep-flatten",
						"flags": "",
						"format": "2",
						"name": "mmrbd1",
						"order": "22",
						"pool_id": "4",
						"size": "128",
						"block_name_prefix": "rbd_data.1e3f238e1f29"
					}
				},
				"erasure_code_profile": "",
				"pg_num": "128",
				"pool_id": "4",
				"pool_name": "test_pool"
			}
		}
	}]
}
 Here , you can see "rbds" parameter inside cluster/pools.
so it means we have to go through the for loop only inside rbds for each cluster and 
also we need to have cluster details (id, name ) for specific pool (which is basically inside cluster) . we should have to put 3 level for loop 

1) first for loop - for cluster iteration
2) second for loop - for pool iteration
3) third for loop - for rbds iteration